File: android_client/android/app/src/main/java/ru/smarthome/database/store/LinkedAccountsStorage.java
package ru.smarthome.database.store;

import com.google.firebase.database.DatabaseReference;

import ru.smarthome.database.model.LinkedAccounts;
import ru.smarthome.database.store.listeners.LinkedAccountsListener;

public interface LinkedAccountsStorage {

    void postLinkedAccounts(LinkedAccounts linkedAccounts);

    void postLinkedAccounts(LinkedAccounts linkedAccounts, DatabaseReference.CompletionListener listener);

    void getLinkedAccounts(LinkedAccountsListener listener);

}

File: android_client/android/app/src/main/java/ru/smarthome/database/store/LinkedAccountsStorage.kt
package ru.smarthome.database.store

import com.google.firebase.database.DatabaseReference

import ru.smarthome.database.model.LinkedAccounts
import ru.smarthome.database.store.listeners.LinkedAccountsListener

interface LinkedAccountsStorage {
    fun postLinkedAccounts(linkedAccounts: LinkedAccounts)
    fun postLinkedAccounts(linkedAccounts: LinkedAccounts, listener: DatabaseReference.CompletionListener)
    fun getLinkedAccounts(listener: LinkedAccountsListener)

}
